extra_data/read_machinery.py

```diff
@@ -2,15 +2,17 @@
 
 The public API is in extra_data.reader; this is internal code.
 """
-from glob import iglob
 import logging
 import math
-import numpy as np
 import os.path as osp
 import re
 import time
+from collections import defaultdict
+from glob import iglob
+from typing import Union
```

Comment:
```suggestion
from glob import iglob
```

LGTM pointed out that there were a couple of unused imports.